File: data/Y22/day1/day1.go
package Y22

import (
	"adventofcode/src-go/helpers"
	"sort"
	"strconv"
)

func Day1() helpers.DayResults {
	day1Example := helpers.ReadFileByLine("data/Y22/day1/example.txt")
	day1 := helpers.ReadFileByLine("data/Y22/day1/data.txt")

	calorieCount := calculateCalories(day1Example)
	part1Example := calorieCount[len(calorieCount)-1]
	part2Example := calorieCount[len(calorieCount)-1] + calorieCount[len(calorieCount)-2] + calorieCount[len(calorieCount)-3]

	calorieCount = calculateCalories(day1)
	part1Result := calorieCount[len(calorieCount)-1]
	part2Result := calorieCount[len(calorieCount)-1] + calorieCount[len(calorieCount)-2] + calorieCount[len(calorieCount)-3]

	results := helpers.DayResults{
		Day:          1,
		Year:         2022,
		Part1Example: part1Example,
		Part1:        part1Result,
		Part2Example: part2Example,
		Part2:        part2Result,
	}

	return results
}

func calculateCalories(data []string) []int {
	var calorieCount []int
	total := 0
	for _, line := range data {
		if line == "" {
			calorieCount = append(calorieCount, total)
			total = 0
			continue
		}

		val, err := strconv.Atoi(line)
		if err != nil {
			panic(err)
		}

		total += val
	}

	sort.Slice(calorieCount, func(i, j int) bool {
		return calorieCount[i] < calorieCount[j]
	})

	return calorieCount
}

File: data/Y22/day1/Day1.php
<?php

namespace trizzssdddAdventOfCode\Y22;

use trizz\AdventOfCode\Solution;

final class Day1 extends Solution
{
    public static null|int|string $part1ExampleResult = 24000;

    public static null|int|string $part1Result = 72240;

    public static null|int|string $part2ExampleResult = 45000;

    public static null|int|string $part2Result = 210957;

    public bool $filterDataOnLoad = false;

    #[\Override]
    public function part1(array $data): int
    {
        return $this->calculateCalories($data)[0];
    }

    #[\Override]
    public function part2(array $data): int
    {
        $results = $this->calculateCalories($data);

        return $results[0] + $results[1] + $results[2];
    }

    /**
     * @param string[] $data
     *
     * @return int[]
     */
    private function calculateCalories(array $data): array
    {
        $results = [];
        $tmpResult = 0;
        foreach ($data as $value) {
            if ($value !== '') {
                $tmpResult += (int) $value;

                continue;
            }

            $results[] = $tmpResult;
            $tmpResult = 0;
        }

        rsort($results);

        return $results;
    }
}
